### PR TITLE
Patch PVCs when storage request is increased

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
fixes https://github.com/pingcap/tidb-operator/issues/3004
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Support patching PVCs when the storage request is increased. Due to the limitation of the current statefulset, new PVCs will be created with the old storage request and resized later
```

### CODE REVIEW

**Comment by cofyc** [comment]
cc @mightyguava

File: pkg/manager/member/pvc_resizer_test.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+						},
+					},
+				},
+			},
+			sc: newStorageClass("sc", false),
+			pvcs: []*v1.PersistentVolumeClaim{
+				newPVCWithStorage("pd-0", label.PDLabelVal, "sc", "2Gi"),
+			},
+			wantPVCs: []*v1.PersistentVolumeClaim{
+				newPVCWithStorage("pd-0", label.PDLabelVal, "sc", "2Gi"),
```
Comment (mightyguava) [inline pkg/manager/member/pvc_resizer_test.go:277]: nice test!

**Comment by cofyc** [comment]
> can/should the operator also monitor that the volume expansion is eventually applied? I guess it can only do that if ExpandInUsePersistentVolumes is enabled, which it probably can't check

it can but the operator can only report some events, but users can watch the PVCs too. as for now, I think it's not necessary.

File: pkg/manager/member/pvc_resizer.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+// Limitations:
+//
+// - Note that the current statfulset implementation does not allow
+//   `volumeClaimTemplates` to be changed, so new PVCs created by statefulset
+//   controller will use the old storage request.
+// - This is best effort, before statefulset volume resize feature (e.g.
+//   https://github.com/kubernetes/enhancements/pull/1848) to be implemented.
+// - If the feature `ExpandInUsePersistentVolumes` is not enabled or the volume
+//   plugin does not support, the pod referencing the volume must be deleted and
+//   recreted after the `FileSystemResizePending` condition becomes true.
```
Comment (DanielZhangQD) [inline pkg/manager/member/pvc_resizer.go:61]:
```suggestion
//   recreated after the `FileSystemResizePending` condition becomes true.
```

**Pull request merged**